homeassistant/components/overkiz/cover.py

```diff
@@ -1,40 +1,413 @@
+            await self.executor.async_execute_command(command)
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        if command := self.entity_description.close_command:
+            await self.executor.async_execute_command(command)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        if command := self.entity_description.stop_command:
+            await self.executor.async_execute_command(command)
+
+    @property
+    def current_cover_tilt_position(self) -> int | None:
+        """Return current position of cover tilt.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        state_name = self.entity_description.current_tilt_position_state
+
+        if (
+            state_name
+            and (state := self.device.states[state_name])
+            and (position := state.value_as_int)
+        ):
+            if self.entity_description.invert_position:
+                position = 100 - position
+
+            return position
+
+        return None
+
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
+        """Move the cover tilt to a specific position."""
+        position = kwargs[ATTR_TILT_POSITION]
+        if command := self.entity_description.set_tilt_position_command:
+            await self.executor.async_execute_command(command, position)
+
+    async def async_open_tilt_cover(self, **kwargs: Any) -> None:
+        """Open the cover tilt."""
+        if command := self.entity_description.open_tilt_command:
+            await self.executor.async_execute_command(command)
+
+    async def async_close_tilt_cover(self, **kwargs: Any) -> None:
+        """Close the cover tilt."""
+        if command := self.entity_description.close_tilt_command:
+            await self.executor.async_execute_command(command)
+
+    async def async_stop_tilt_cover(self, **kwargs: Any) -> None:
+        """Stop the cover tilt."""
+        if command := self.entity_description.stop_tilt_command:
+            await self.executor.async_execute_command(command)
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+        if command := self.entity_description.open_command:
+            if self.is_running(command):
+                return True
+
+        if self.moving_offset is None:
+            return None
+
+        if self.entity_description.invert_position:
+            return self.moving_offset > 0
+        return self.moving_offset < 0
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is opening or not."""
+        if command := self.entity_description.close_command:
+            if self.is_running(command):
+                return True
+
+        if self.moving_offset is None:
+            return None
+
+        if self.entity_description.invert_position:
```

Comment:
Maybe the moving_offset could do the invert position, and it will simplify the code in is_closing and is_opening